[cmd/opampsupervisor] Flaky E2E test: TestSupervisorRemoteConfigApplyStatus

### Component(s)

cmd/opampsupervisor

### Describe the issue you're reporting

This test seems to randomly panic in PRs with the following trace:
```
panic: interface conversion: interface {} is nil, not *protobufs.RemoteConfigStatus

goroutine 1711 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/cmd/opampsupervisor.TestSupervisorRemoteConfigApplyStatus.func1.2(0x16e8be0?)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/e2e_test.go:2121 +0x132
github.com/stretchr/testify/assert.EventuallyWithT.func1()
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:2096 +0x70
created by github.com/stretchr/testify/assert.EventuallyWithT in goroutine 1657
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:2108 +0x1a5
exit status 2
```

Here are links to some recent failure logs:
- [failure 1](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/18883962260/job/53895518175?pr=43845)
- [failure 2](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/18855068789/job/53801518421)
- [failure 3](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/18850258766/job/53785662198)
- [failure 4](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/18826120613/job/53709504389)

Based on the test code and error, I suspect fixing this is just a matter of adding a `require.NotNil` before casting the `any`.

It looks like this flakiness was first mitigated in #42551, and #42654 removed the mitigation without a proper fix.

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

--- a/cmd/opampsupervisor/e2e_test.go
+++ b/cmd/opampsupervisor/e2e_test.go
@@ -2118,7 +2118,9 @@ func TestSupervisorRemoteConfigApplyStatus(t *testing.T) {
 
 			// Check that the status is set to APPLYING
 			require.EventuallyWithT(t, func(c *assert.CollectT) {
-				status := remoteConfigStatus.Load().(*protobufs.RemoteConfigStatus)
+				statusVal := remoteConfigStatus.Load()
+				require.NotNil(c, statusVal) // not set yet
+				status := statusVal.(*protobufs.RemoteConfigStatus)
 				t.Log("status", status.Status)
 				assert.Equal(c, protobufs.RemoteConfigStatuses_RemoteConfigStatuses_APPLYING, status.Status)
 			}, 5*time.Second, 100*time.Millisecond)

